packages/keyvault/lib/objectIdentifier.ts

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+import { format } from "util";
+import { parse } from "url";
+
+export class ObjectIdentifier {
+    /** The vault URI.
+     * @member {string}
+     */
+    vault: string;
+
+    /** The key name.
+      * @member {string}
+      */
+    name: string;
+
+    /** The key version. May be undefined.
+     * @member {string}
+     */
+    version?: string;
+
+    /** The base identifier (i.e. without the version).
+     * @member {string}
+     */
+    baseIdentifier: any;
+
+    /** The full identifier if a version was informed; otherwise is the same value of baseIdentifier.
+     * @member {string}
+     */
+    identifier: any;
+
+    /**
+     * Creates an ObjectIdentifier object for a key.
+     * @param {string} collection The collection name.
+     * @param {string} vault The vault URI.
+     * @param {string} name The key name.
+     * @param {string} [version=undefined] The object version.
+     * @return {ObjectIdentifier} An object that represents the key identifier.
+     */
+    constructor(collection: string, vault: string, name: string, version?: string) {
+        this.vault = vault;
+        this.name = name;
+        this.version = version;
+        this.baseIdentifier = format("%s/%s/%s", vault, collection, name);
```

Comment:
Suggest using backtick template strings instead of taking this dependency e.g. `${vault}/${collection}/${name}`